src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -2201,36 +2209,36 @@ class ObservationStore(
             } else {
               null
             }
+        val survivalRateDenominator =
+            DSL.field(
+                DSL.select(DSL.sum(PLOT_T0_DENSITY.PLOT_DENSITY))
+                    .from(PLOT_T0_DENSITY)
+                    .where(survivalRateDensityCondition)
+                    .and(PLOT_T0_DENSITY.SPECIES_ID.eq(speciesKey.id))
+            )
+
+        val survivalRate =
+            if (isPermanent && speciesKey.id != null) {
+              DSL.`val`(totalLive).mul(100).div(survivalRateDenominator)
+            } else {
+              DSL.inline(null, SQLDataType.INTEGER)
+            }
 
         val rowsInserted =
             dslContext
-                .insertInto(
-                    table,
-                    observationIdField,
-                    scopeIdField,
-                    certaintyField,
-                    speciesIdField,
-                    speciesNameField,
-                    totalLiveField,
-                    totalDeadField,
-                    totalExistingField,
-                    cumulativeDeadField,
-                    permanentLiveField,
-                    mortalityRateField,
-                )
-                .values(
-                    observationId,
-                    scopeId,
-                    speciesKey.certainty,
-                    speciesKey.id,
-                    speciesKey.name,
-                    totalLive,
-                    totalDead,
-                    totalExisting,
-                    cumulativeDead,
-                    permanentLive,
-                    mortalityRate,
-                )
+                .insertInto(table)
+                .set(observationIdField, observationId)
+                .set(scopeIdField, scopeId)
+                .set(certaintyField, speciesKey.certainty)
+                .set(speciesIdField, speciesKey.id)
+                .set(speciesNameField, speciesKey.name)
+                .set(totalLiveField, totalLive)
+                .set(totalDeadField, totalDead)
+                .set(totalExistingField, totalExisting)
+                .set(cumulativeDeadField, cumulativeDead)
+                .set(permanentLiveField, permanentLive)
+                .set(mortalityRateField, mortalityRate)
+                .set(survivalRateField, survivalRate)
```

Comment:
Had to change to `.set` since jooq doesn't allow a mix of raw kotlin values and DSL values

Reply:
Right, I knew that was possible, but it felt cleaner to use `.set` than 11 instances of `DSL.value`